EpikCord/__init__.py

```diff
@@ -152,29 +154,16 @@ def register_slash_command(func):
 
         return register_slash_command
 
-    def user_command(self, name: Optional[str] = None):
-        def register_slash_command(func):
-            result = ClientUserCommand(
-                **{
-                    "callback": func,
-                    "name": name or func.__name__,
-                }
-            )
-            self.commands[name](result)
-            return result
+    def _register_command(self, command_type, name, func):
+        command_instance = command_type(callback=func, name=name or func.__name__)
+        self.commands[name] = command_instance
+        return command_instance
 
-        return register_slash_command
+    def user_command(self, name: Optional[str] = None):
```

Comment:
Where does this get func from?

Reply:
from `_register_command`

Reply:
No how do you get the callback?

Reply:
Oh right, so the return acts like the decorators `return wrapper/inner/whatever`